poetry/packages/package.py

```diff
@@ -340,7 +340,7 @@ def to_dependency(self):
 
         name = "{} (=={})".format(self._name, self._version)
 
-        if not self.marker.is_any():
+        if not self.marker.is_any() and not self.marker.is_empty():
```

Comment:
That's not what we want here. An empty marker is a marker that will always evaluate to false. By dropping it we make the dependency always valid which is contrary to what an empty marker means.

The issue is that we should not have empty markers make it to the lock file, which is the root cause of the problem you are seeing and this is where the underlying issue should be fixed.

Reply:
Would it be sufficient to just make sure they don't get serialized to the lock file, or is there something more fundamental about dependency resolution that needs to prune them?